app/Services/CampaignService.php

```diff
@@ -176,4 +177,33 @@ public function getCampaignIdsFromSignups()
 
         return $ids ? $ids : null;
     }
+
+    /**
+     * Appends status counts to a collection of campaigns.
+     *
+     * @return Illuminate\Database\Eloquent\Collection $campaigns
+     */
+    public function getCampaignPostStatusCounts($campaigns)
+    {
+        $ids = array_pluck($campaigns, 'id');
+
+        $campaignsWithSignups = Signup::campaign($ids)
+            ->includePostStatusCounts()
+            ->get()
+            ->groupBy('campaign_id');
```

Comment:
I think we want this `groupBy` above the `get` so it's grouping in the SQL query (and only returning the grouped results), rather than in-memory using the returned Collection instance.